### PR TITLE
Feature/agent groups

### PR DESCRIPTION
**Pull request by toabi**
This makes it possible to have different agent groups with different instance types. So it's possible to have mixed type instances.

### CODE REVIEW

File: manifests/hello-kubernetes.yaml [hunk 1]
```diff
@@ -14,39 +14,66 @@ spec:
...
 kind: Service
 metadata:
   name: hello-kubernetes
   annotations:
-   load-balancer.hetzner.cloud/location: nbg1
+    load-balancer.hetzner.cloud/location: nbg1
+    load-balancer.hetzner.cloud/use-private-ip: "true"
 spec:
-  type: LoadBalancer
+  type: ClusterIP
```
Comment (StarpTech) [inline manifests/hello-kubernetes.yaml]: The example should be used with hetzner load balancer.

File: manifests/hello-kubernetes.yaml [hunk 2]
```diff
@@ -14,39 +14,66 @@ spec:
...
-kind: PersistentVolumeClaim
+apiVersion: networking.k8s.io/v1
+kind: Ingress
 metadata:
-  name: csi-pvc
+  name: hello-kubernetes
+  annotations:
+    cert-manager.io/cluster-issuer: "letsencrypt"
 spec:
-  accessModes:
```
Comment (StarpTech) [inline manifests/hello-kubernetes.yaml:47]: Same here. The example should demonstrate hetzner CSI usage.

File: manifests/hello-kubernetes.yaml [hunk 3]
```diff
@@ -14,39 +14,66 @@ spec:
...
-  - ReadWriteOnce
-  resources:
-    requests:
-      storage: 10Gi
-  storageClassName: hcloud-volumes
\ No newline at end of file
+  tls:
+  - hosts:
+    - hello.toabi.mecodia.cloud
```
Comment (StarpTech) [inline manifests/hello-kubernetes.yaml]: This won't work for everybody. I'd remove tls configuration from the example.

**Review by StarpTech — CHANGES_REQUESTED** [review]

File: manifests/hello-kubernetes.yaml (hunk 3 above)
Comment (toabi) [reply manifests/hello-kubernetes.yaml]: Oh sorry didn't want to commit this example at all :D

**Review by StarpTech — APPROVED** [review]
LGTM

**Pull request merged**